test-dts/defineComponent.test-d.tsx

```diff
@@ -43,6 +43,71 @@ describe('with object props', () => {
     date?: Date
   }
 
+  type ExpectedPropsType = {
```

Comment:
this might might a bit too much to maintain, since this is duplicated from the actual `props`, it would be better if the props definition was placed outside as `const` and be used to check the type

Reply:
👌